src/lib/core/tools.c

```diff
@@ -817,9 +858,18 @@ on_object_weak_unref (gpointer user_data, GObject * obj)
 guint
 bt_g_object_idle_add (GObject * obj, gint pri, GSourceFunc func)
 {
-  guint id = g_idle_add_full (pri, func, obj, NULL);
-
-  g_object_weak_ref (obj, on_object_weak_unref, GUINT_TO_POINTER (id));
+  g_assert (bt_g_object_idle_obj); // bt_init not called
+  
+  guint id =
+    g_idle_add_full (pri, func, obj,
+        bt_g_object_idle_on_callback_removed);
+
+  g_hash_table_insert (bt_g_object_idle_obj,
```

Comment:
I think this is fine, but do you see any issues wrt to thread safety? When inserting into the hashtable it looks fine, just wonder about the multiple remove calls.